src/main/org/apache/tools/ant/taskdefs/optional/junitlauncher/JUnitLauncherTask.java

```diff
@@ -0,0 +1,508 @@
+        sysOutStreamer.setDaemon(true);
+        sysOutStreamer.setName("junitlauncher-sysout-stream-reader");
+        sysOutStreamer.setUncaughtExceptionHandler((t, e) -> this.log("Failed in sysout streaming", e, Project.MSG_INFO));
+        sysOutStreamer.start();
+        return Optional.of(new SwitchedStreamHandle(pipedOutputStream, streamer));
+    }
+
+    private Optional<SwitchedStreamHandle> trySwitchSysErr(final TestRequest testRequest) {
+        if (!testRequest.interestedInSysErr()) {
+            return Optional.empty();
+        }
+        final PipedOutputStream pipedOutputStream = new PipedOutputStream();
+        final PipedInputStream pipedInputStream;
+        try {
+            pipedInputStream = new PipedInputStream(pipedOutputStream);
+        } catch (IOException ioe) {
+            // log and return
+            return Optional.empty();
+        }
+
+        final PrintStream printStream = new PrintStream(pipedOutputStream, true);
+        System.setErr(new PrintStream(printStream));
+
+        final SysOutErrStreamReader streamer = new SysOutErrStreamReader(this, pipedInputStream,
+                StreamType.SYS_ERR, testRequest.getSysErrInterests());
+        final Thread sysErrStreamer = new Thread(streamer);
+        sysErrStreamer.setDaemon(true);
+        sysErrStreamer.setName("junitlauncher-syserr-stream-reader");
+        sysErrStreamer.setUncaughtExceptionHandler((t, e) -> this.log("Failed in syserr streaming", e, Project.MSG_INFO));
+        sysErrStreamer.start();
+        return Optional.of(new SwitchedStreamHandle(pipedOutputStream, streamer));
+    }
+
+    private static void safeClose(final Closeable... closeables) {
+        for (final Closeable closeable : closeables) {
+            try {
+                closeable.close();
+            } catch (Exception e) {
+                // ignore
+            }
+        }
+    }
+
+    private enum StreamType {
+        SYS_OUT,
+        SYS_ERR
+    }
+
+    private static final class SysOutErrStreamReader implements Runnable {
+        private static final byte[] EMPTY = new byte[0];
+
+        private final JUnitLauncherTask task;
+        private final InputStream sourceStream;
+        private final StreamType streamType;
+        private final Collection<TestResultFormatter> resultFormatters;
+        private volatile SysOutErrContentDeliverer contentDeliverer;
+
+        SysOutErrStreamReader(final JUnitLauncherTask task, final InputStream source, final StreamType streamType, final Collection<TestResultFormatter> resultFormatters) {
+            this.task = task;
+            this.sourceStream = source;
+            this.streamType = streamType;
+            this.resultFormatters = resultFormatters == null ? Collections.emptyList() : resultFormatters;
+        }
+
+        @Override
+        public void run() {
+            if (this.resultFormatters.isEmpty()) {
+                // no one to feed the stream content to
```

Comment:
I believe this may cause the `PipedOutputStream` to consider the `PipedInputStream` "broken" soon and throw on any `write` operation.

Reply:
This check is actually "dead code", in the sense that this will never be true. I wanted to avoid running these threads when there's no result formatter interested in the sysout/syserr content. But that check obviously needs to happen before the thread is even created and in fact, there's already such a check in the `trySwitchSysOut` and `trySwitchSysErr` methods (the place where this thread gets created).

So I've now updated the PR to remove this check.